retrybp/doc_test.go

```diff
@@ -85,3 +86,70 @@ func Example() {
 	// TODO: use result
 	_ = result
 }
+
+// This example demonstrates how to use retrybp with breakerbp.
+func ExampleBreakerErrorFilter() {
+	const timeout = time.Millisecond * 200
+
+	// TODO: use the actual type of your operation.
+	type resultType = int
+
+	// In real code this should be your actual operation you need to retry.
+	// Here we just use a placeholder for demonstration purpose.
+	operation := func() (resultType, error) {
+		return 0, nil
+	}
+
+	breaker := breakerbp.NewFailureRatioBreaker(breakerbp.Config{
+		// TODO: Add breaker configs.
+	})
+
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	const defaultRetries = 2
+	var result resultType
+	err := retrybp.Do(
+		ctx,
+		func() error {
+			// This is the main function call,
+			// wrapped with breaker.Execute to apply circuit breaker on it.
+			res, err := breaker.Execute(func() (interface{}, error) {
+				ret, err := operation()
+				return ret, err
+			})
+			result, _ = res.(resultType)
```

Comment:
this is a bad pattern:
* don't leak bad values into outer scopes (i.e. ensure err != nil first before assigning)
* don't discard errors (check the OK value or let it panic)